### PR TITLE
Add Initial Experiment Step

### PR DESCRIPTION
**Pull request by dthomson25**

### CODE REVIEW

**Comment by dthomson25** [comment]
Initial PR to resolve https://github.com/argoproj/argo-rollouts/issues/92

File: pkg/apis/rollouts/v1alpha1/types.go [hunk 1]
```diff
@@ -144,6 +144,48 @@ type CanaryStrategy struct {
 	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
 }
 
+// RolloutCanaryExperimentStep defines a template that is used to create a experiment for a step
+type RolloutCanaryExperimentStep struct {
```
Comment (jessesuen) [inline pkg/apis/rollouts/v1alpha1/types.go]: I can imagine a future where ExperimentStep is not just being done as part of Canary, but BlueGreen. May want to name this just RolloutExperimentStep to not box ourselves.

File: pkg/apis/rollouts/v1alpha1/types.go [hunk 2]
```diff
@@ -144,6 +144,48 @@ type CanaryStrategy struct {
...
+	SpecRef ReplicaSetSpecRef `json:"specRef"`
+	// Replicas replica count for the template
+	Replicas int32 `json:"replicas"`
+	// AdditionalSelectors additional selectors to use for the RS created from the template
+	// +optional
+	Metadata RolloutExperimentAdditionalMetadata `json:"metadata,omitempty"`
+}
+
+//RolloutExperimentAdditionalMetadata extra labels to add to the template
+type RolloutExperimentAdditionalMetadata struct {
```
Comment (jessesuen) [inline pkg/apis/rollouts/v1alpha1/types.go]: How about: `PodTemplateMetadata`. I would make clear that this metadata will go into the labels/annotations of the actual pod

File: pkg/apis/rollouts/v1alpha1/types.go (hunk 1 above)
Comment (dthomson25) [reply pkg/apis/rollouts/v1alpha1/types.go]: Done

File: pkg/apis/rollouts/v1alpha1/types.go (hunk 2 above)
Comment (dthomson25) [reply pkg/apis/rollouts/v1alpha1/types.go]: done

**Pull request merged**